core/templates/pages/topic-editor-page/rearrange-skills-in-subtopics-modal.controller.spec.ts

```diff
@@ -116,4 +123,25 @@ describe('Rearrange Skills In Subtopic Modal Controller', function() {
     expect(ctrl.editableName).toEqual('');
     expect(ctrl.selectedSubtopicId).toEqual(0);
   });
+
+  it('should call initEditor on calls from topic being initialized',
+    function() {
+      topicInitializedEventEmitter = new EventEmitter();
+      topicReinitializedEventEmitter = new EventEmitter();
+
+      spyOnProperty(TopicEditorStateService, 'onTopicInitialized').and.callFake(
+        function() {
+          return topicInitializedEventEmitter;
+        });
+      spyOnProperty(
+        TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+        function() {
+          return topicReinitializedEventEmitter;
+        });
+      spyOn(ctrl, 'initEditor').and.callThrough();
+      ctrl.init();
+      topicInitializedEventEmitter.emit();
+      topicReinitializedEventEmitter.emit();
+      expect(ctrl.initEditor).toHaveBeenCalledTimes(3);
```

Comment:
Just a suggestion, can you add an assertion between each call so that we know each of these call the function once? (It is theoretically possible that this test would pass if one of them called it twice or thrice and the other doesn't call it).

Reply:
Done